draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -130,51 +130,77 @@ Perhaps this could be included as a svc="baz" parameter
-machines located at svc2.example.net with TCP port 8002 and HTTP/3 is
-available on a cluster of machines located at svc3.example.net with
-UDP port 8003.  The client can use the specified ESNI keys to encrypt
-the SNI values of "example.com" and "www.example.com" in the handshake
-with these alternative service endpoints.  When connecting, clients will
-continue to treat the authoritative origins as "https://example.com"
-and "https://www.example.com", respectively.
-
-For services other than HTTPS (as well as for HTTPS origins
-with non-default ports), the SVCB RR and an {{?Attrleaf}} label will be used.
-For example, to reach an example resource of
+    simple.example. 300 IN A    192.0.2.1
+                           AAAA 2001:db8::1
+
+The domain owner could publish new records like
+
+    simple.example. 7200 IN HTTPSSVC 1 . alpn=h3 ...
+                            HTTPSSVC 2 . alpn=h2 ...
+
+The presence of these records indicates to clients that simple.example
+supports HTTPS, and the key=value pairs indicate that it prefers HTTP/3
+but also supports HTTP/2.  The records can also include other information
+(e.g. non-standard ports, ESNI configuration).
+
+## Example: Apex aliasing
+
+Consider a zone that is using CNAME aliasing:
+
+    $ORIGIN aliased.example. ; A zone that is using a hosting service
+    ; Subdomain aliased to a high-performance server pool
+    www             7200 IN CNAME pool.svc.example.
+    ; Apex domain on fixed IPs because CNAME is not allowed at the apex
+    aliased.example. 300 IN A     192.0.2.1
+                         IN AAAA  2001:db8::1
+
+With HTTPSSVC, the owner of aliased.example could alias the apex by
+adding one additional record:
+
+    aliased.example. 7200 IN HTTPSSVC 0 pool.svc.example.
+
+With this record in place, HTTPSSVC-aware clients will use the same
+server pool for complex.example and www.complex.example.  (They will
+also upgrade to HTTPS on complex.example.)  Non-HTTPSSVC-aware clients
+will just ignore the new record.
+
+## Example: Parameter binding
+
+Suppose that svc.example's default server pool supports HTTP/2, and
+it has deployed HTTP/3 on a new server pool with a different
+configuration.  This can be expressed in the following form:
+
+    $ORIGIN svc.example. ; A hosting provider.
+    pool 7200 IN HTTPSSVC 1 h3 alpn=h3 esniconfig="123..."
+                 HTTPSSVC 2 .  alpn=h2 esniconfig="abc..."
+    pool  300 IN A        192.0.2.2
+                 AAAA     2001:db8::2
+    h3    300 IN A        192.0.2.3
+                 AAAA     2001:db8::3
+
+This configuration is entirely compatible with the "Apex aliasing" example,
+whether the client supports HTTPSSVC or not.  If the client does support
+HTTPSSVC, all connections will be upgraded to HTTPS, and clients will
+use HTTP/3 if they can.  Parameters are "bound" to each server pool, so
+each server pool can have its own protocol, ESNI configuration, etc.
```

Comment:
Perhaps explicitly call out that pool.svc.example. may be either the target of a CNAME chain or an alias?
For example:

$ORIGIN example.com.
www    3600 IN CNAME pool.svc.example.
 .          3600 IN HTTPSSVC 0 pool.svc.example.

Reply:
It would also be good to clarify that neither form impacts the domain name in the origin.
For example:
------

Similar to CNAME, HTTPSSVC aliases have no impact on the origin name.  When connecting, clients will continue to treat the authoritative origins as "https://www.example.com" and "https://example.com", respectively, and will validate TLS server certificates accordingly.

Reply:
> Perhaps explicitly call out that pool.svc.example. may be either the target of a CNAME chain or an alias?

This is already in the preceding example.

> It would also be good to clarify that neither form impacts the domain name in the origin.

Added.